consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/CliqueForksSchedulesFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.clique;
+
+import org.hyperledger.besu.config.CliqueConfigOptions;
+import org.hyperledger.besu.config.CliqueFork;
+import org.hyperledger.besu.config.GenesisConfigOptions;
+import org.hyperledger.besu.consensus.common.ForkSpec;
+import org.hyperledger.besu.consensus.common.ForksSchedule;
+import org.hyperledger.besu.consensus.common.ForksScheduleFactory;
+
+/** The Clique forks schedules factory. */
+public class CliqueForksSchedulesFactory {
+
+  /**
+   * Create forks schedule.
+   *
+   * @param genesisConfig the genesis config
+   * @return the forks schedule
+   */
+  public static ForksSchedule<CliqueConfigOptions> create(
+      final GenesisConfigOptions genesisConfig) {
+    return ForksScheduleFactory.create(
+        genesisConfig.getCliqueConfigOptions(),
+        genesisConfig.getTransitions().getCliqueForks(),
+        CliqueForksSchedulesFactory::createCliqueConfigOptions);
+  }
+
+  private static CliqueConfigOptions createCliqueConfigOptions(
+      final ForkSpec<CliqueConfigOptions> lastSpec, final CliqueFork fork) {
+    final MutableCliqueConfigOptions cliqueConfigOptions =
+        new MutableCliqueConfigOptions(lastSpec.getValue());
+
+    fork.getBlockPeriodSeconds().ifPresent(cliqueConfigOptions::setBlockPeriodSeconds);
+
+    return cliqueConfigOptions;
```

Comment:
From what I see, there should not be a need for this mutable object, since if you use _immutables_ annotation for `CliqueConfigOptions` then you can do something like this

```
var next = ImmutableCliqueConfigOptions.builder().from(lastSpec.getValue());
fork.getBlockPeriodSeconds().ifPresent(next::blockPeriodSeconds);
return next.build();
```

and avoid having mutable objects around, since the conf should newer change once it is created.

Reply:
This Mutable*ConfigOptions approach is pretty ingrained in the transition functionality used throughout the codebase, ie for BFT too. Can spike this out unless @jframe already knows there would be an issue with this?

Reply:
Should be fine to change it using a builder. It's only mutable for building the CliqueConfigOptions from the transitions.

Reply:
done this for Clique, but not BFT

Reply:
fine, the other part could be done later